kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -173,6 +201,21 @@ private void handleMetadata(ColumnVector metadataVector) {
         }
     }
 
+    /**
+     * Get the commit info from the winning transactions.
+     *
+     * @param commitInfoVector commit info rows from the winning transactions
+     * @return the commit info
+     */
+    private CommitInfo getCommitInfo(ColumnVector commitInfoVector) {
```

Comment:
Return `Optional<CommitInfo>` directly, rather than using `Optional.ofNullable` at the caller.